src/lib/replication/service/repl_service_impl.cpp

```diff
@@ -74,7 +73,7 @@ ReplicationServiceImpl::open_repl_dev(uuid_t group_id, std::unique_ptr< ReplDevL
     auto it = m_rd_map.find(group_id);
     if (it != m_rd_map.end()) {
         // We already loaded the ReplDev, just call the group_id and attach the listener
-        auto& repl_dev = it->second;
+        auto repl_dev = it->second;
```

Comment:
I am not clear how is this fixes the problem. At this time, someone already opened the repl_dev and group_id -> repl_dev has been mapped. So how is this storing by referencing or incrementing the ref_count of repl_dev solves this?

Reply:
but at Line80, this repl_dev will be moved, that is why I changed to make a copy of shared_ptr